### PR TITLE
CPM-65;Feat;Adds default expansion level

### PR DESCRIPTION
**Pull request by HamedRajabiVaramini**
Adds Default expansion level feature

## Changes
- Adds a new option 'Default Expansion Level' to Tree Panel 
- Applies default expansion level on loading module

### CODE REVIEW

File: src/components/TreePanel.tsx [hunk 1]
```diff
@@ -409,19 +413,23 @@ function transformData(rows: string[]): TreeNodeData[] {
...
         // find parent level
-        const parent = items.find((i) => i.id === row[j - 1].id) ?? throwExpression('parent not found')
+        const parent = items.find((i) => i.id === row[levelIndex - 1].id) ?? throwExpression('parent not found')
         if (!parent.children) {
           parent.children = []
         }
         items = parent.children
         item.parent = parent
       }
+      const levelNumber = levelIndex + 1
```
Comment (jackieli-tes) [inline src/components/TreePanel.tsx]: probably use 0 base index which is what developers are more familiar with

File: src/module.ts [hunk 2]
```diff
@@ -39,6 +39,12 @@ export const plugin = new PanelPlugin<TreeOptions>(TreePanel).setPanelOptions((b
       description: 'Format selection into query from a handlebars.js template',
       defaultValue: defaultFormatTemplate,
     })
+    .addNumberInput({
+      path: 'defaultExpansionLevel',
+      name: 'Default Expansion Level',
+      description: 'The default level value to expand tree at first look',
+      defaultValue: 6,
```
Comment (jackieli-tes) [inline src/module.ts]: default value is probably too high, we can always configure it to our requirement in panel config. Use 0 or 1.

**Review by jackieli-tes — CHANGES_REQUESTED** [review]

File: src/components/TreePanel.tsx (hunk 1 above)
Comment (HamedRajabiVaramini) [reply src/components/TreePanel.tsx]: No problem

File: src/module.ts (hunk 2 above)
Comment (HamedRajabiVaramini) [reply src/module.ts]: If we set this parameter to zero, users just consider the `Asia` node, it seems we didn't do anything. I discussed with @alializadehtarget and he sent me this:
image 
So `defaultValue: 3` seems desired option for him.

**Review by jackieli-tes — APPROVED** [review]

**Pull request merged**